atomicfile/atomicfile.go

```diff
@@ -10,17 +10,30 @@ package atomicfile // import "tailscale.com/atomicfile"
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 )
 
 // WriteFile writes data to filename+some suffix, then renames it
 // into filename.
 func WriteFile(filename string, data []byte, perm os.FileMode) error {
 	tmpname := filename + ".new.tmp"
-	if err := ioutil.WriteFile(tmpname, data, perm); err != nil {
-		return fmt.Errorf("%#v: %v", tmpname, err)
+
+	f, err := os.OpenFile(tmpname, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, perm)
+	if err != nil {
+		return err
+	}
+	if _, err = f.Write(data); err != nil {
+		f.Close()
+		return err
+	}
+	if err = f.Sync(); err != nil {
+		f.Close()
+		return err
 	}
+	if err = f.Close(); err == nil {
```

Comment:
you meant `err != nil ` here.

But I sent my counterproposal of this PR in https://github.com/tailscale/tailscale/pull/558 with this bug fixed.